students/dowoon/users/views.py

```diff
@@ -1,3 +1,39 @@
-from django.shortcuts import render
+import json
+
+from django.http import JsonResponse
+from django.views import View
+
+from users.models import User
+from users.validators import validate_email, validate_password, validate_duplicate
+
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+
+        try:
+            name     = data['name']
+            email    = data['email']
+            password = data['password']
+            phone    = data['phone']
+
+            if email == "" or password == "":
+                raise Exception("ERROR : email or password is blank")
+
+            validate_email(email)
+            validate_password(password)
+            validate_duplicate(email)
+        except KeyError:
+            return JsonResponse({"message" : "KEY_ERROR"}, status=400)
+        except Exception as e:
+            return JsonResponse({"message" : str(e)}, status=400)
```

Comment:
`Exception`으로 모든 에러를 다 처리하는게 아닌 내가 의도한 `ValidationError`만 예외처리 해주는게 올바른 방식입니다.
그리고 ValidationError 같은 `<class 'django.core.exceptions.ValidationError'>` 자체를 string화 시켜 출력하는 것보다, Error 의 속성값을 확인하여 `validators.py`에서 작성한대로 `message`만 출력되게 하는게 좋아보입니다.
`print(e.__dir__())`을 return 위에 넣어줌으로 서버에 어떻게 출력되는지 보시고 어떤 것들을 이용할 수 있는지 파악해보세요~
```suggestion
        except ValidationError as e:
            print(e.__dir__()) # 이 부분은 push 시에 지워주세요.
            return JsonResponse({"message" : e.message}, status=400)
```

Reply:
안에 message가 있었군요! 확인 / 수정했습니다.